### ISSUE TITLE
AVDisplayCriteria from formatDescription not responds DV profile only HDR10

### ISSUE THREAD

**Issue by Alanko5** [kingslay/KSPlayer#633]
**Describe the bug**
displayManager.preferredDisplayCriteria = AVDisplayCriteria(refreshRate: refreshRate, formatDescription: formatDescription)
```
Printing description of formatDescription:
<CMVideoFormatDescription 0x2834f9b60 [0x236974b20]> {
	mediaType:'vide' 
	mediaSubType:'x420' 
	mediaSpecific: {
		codecType: 'x420'		dimensions: 3840 x 2160 
	} 
	extensions: {{
    CVBytesPerRow = 11524;
    CVImageBufferChromaLocationBottomField = Left;
    CVImageBufferChromaLocationTopField = Left;
    CVImageBufferColorPrimaries = "ITU_R_2020";
    CVImageBufferTransferFunction = "SMPTE_ST_2084_PQ";
    CVImageBufferYCbCrMatrix = "ITU_R_2020";
    Version = 2;
}}
}
```

**To Reproduce**
starts video with DV
tvOS switch TV to HDR

**Platform (please complete the following information):**
 - Device: Apple TV 4k 3gen
 - OS: tvOS 17

**Screenshots**
<img width="361" alt="Snímka obrazovky 2023-11-13 o 19 29 59" src="https://github.com/kingslay/KSPlayer/assets/71501260/82afdd62-7f78-4e26-9fb6-2e10c68ab4b1">

<<<< AudioFormatDescription >>>> figAudioFormatDescriptionEnsureRichestDecodableLayout signalled err=-12718 (kCMFormatDescriptionError_ValueNotAvailable) (no formats in format list) at FigAudioFormatDescription.c:2148

(Fig) CMVideoFormatDescriptionCreate signalled err=-12710 (kFigFormatDescriptionError_InvalidParameter) (!codecType) at FigVideoFormatDescription.c:417

**Comment by kingslay** [comment]
不要用 AVDisplayCriteria(refreshRate: refreshRate, formatDescription。继续用我们原有的AVDisplayCriteria(refreshRate: refreshRate, videoDynamicRange: dynamicRange.rawValue) 就可以了。目前formatDescription里面是没有dv信息，所以无法判断是dv

**Comment by Alanko5** [comment]
<img width="780" alt="Create criteria ni KS" src="https://github.com/kingslay/KSPlayer/assets/71501260/cdbdef0d-0c8d-4848-a446-f6d0592fac47">
<img width="773" alt="method not exist" src="https://github.com/kingslay/KSPlayer/assets/71501260/2d895f67-e14b-49d8-80bb-ec6eb2ac8339">

**Comment by kingslay** [comment]
没有函数了。你换成这样写
```

   override func updateVideo(refreshRate: Float, isDovi: Bool, formatDescription: CMFormatDescription?) {
        #if os(tvOS) || os(xrOS)
        guard let displayManager = UIApplication.shared.windows.first?.avDisplayManager,
              displayManager.isDisplayCriteriaMatchingEnabled,
              !displayManager.isDisplayModeSwitchInProgress
        else {
            return
        }
        if let formatDescription {
            if KSOptions.displayCriteriaFormatDescriptionEnabled, #available(tvOS 17.0, *) {
                displayManager.preferredDisplayCriteria = AVDisplayCriteria(refreshRate: refreshRate, formatDescription: formatDescription)
            } else {
                let dynamicRange = isDovi ? .dolbyVision : formatDescription.dynamicRange
                displayManager.preferredDisplayCriteria = AVDisplayCriteria(refreshRate: refreshRate, videoDynamicRange: dynamicRange.rawValue)
            }
        }
        #endif
    }
```
